### ISSUE TITLE
Autoredirect not working correctly for paths containing query parameters

### ISSUE THREAD

**Issue by AeonLucid** [tmenier/Flurl#586]
Here is a test reproducing the issue, this is a slightly modified version of `RedirectTests.can_auto_redirect`.

```csharp
[Test]
public async Task can_auto_redirect() {
    HttpTest
        .RespondWith("", 302, new { Location = "http://redir.com/foo" })
        .RespondWith("", 302, new { Location = "/redir2" })
        .RespondWith("", 302, new { Location = "redir3" })
        .RespondWith("", 302, new { Location = "/redir2?a=b&c=d" })
        .RespondWith("done!");

    var resp = await "http://start.com".PostStringAsync("foo!").ReceiveString();

    Assert.AreEqual("done!", resp);
    HttpTest.ShouldHaveMadeACall().Times(5);
    HttpTest.ShouldHaveCalled("http://start.com").WithVerb(HttpMethod.Post).WithRequestBody("foo!")
        .With(call => call.RedirectedFrom == null);
    HttpTest.ShouldHaveCalled("http://redir.com/foo").WithVerb(HttpMethod.Get).WithRequestBody("")
        .With(call => call.RedirectedFrom.Request.Url.ToString() == "http://start.com");
    HttpTest.ShouldHaveCalled("http://redir.com/redir2").WithVerb(HttpMethod.Get).WithRequestBody("")
        .With(call => call.RedirectedFrom.Request.Url.ToString() == "http://redir.com/foo");
    HttpTest.ShouldHaveCalled("http://redir.com/redir2/redir3").WithVerb(HttpMethod.Get).WithRequestBody("")
        .With(call => call.RedirectedFrom.Request.Url.ToString() == "http://redir.com/redir2");
    HttpTest.ShouldHaveCalled("http://redir.com/redir2?a=b&c=d").WithVerb(HttpMethod.Get).WithRequestBody("")
        .With(call => call.RedirectedFrom.Request.Url.ToString() == "http://redir.com/redir2/redir3");
}
```

If we look into the CallLog, it is obvious why the test did not succeed.

![image](https://user-images.githubusercontent.com/4643257/101999239-56669580-3cdb-11eb-88f7-12ffb2ce5bf2.png)

In a real world application, this causes a 404 because it think the query parameters are part of the route.

**Comment by tmenier** [comment]
This is legit. Thanks for reporting.

**Comment by tmenier** [comment]
Fixed & released in 3.0.1
